cql3/expr/expression.cc

```diff
@@ -863,6 +863,26 @@ expression replace_column_def(const expression& expr, const column_definition* n
         }, expr);
 }
 
+expression replace_token(const expression& expr, const column_definition* new_cdef) {
+    return std::visit(overloaded_functor{
+            [] (bool b){ return expression(b); },
+            [&] (const conjunction& conj) {
+                const auto applied = conj.children | transformed(
+                        std::bind(replace_token, std::placeholders::_1, new_cdef));
+                return expression(conjunction{std::vector(applied.begin(), applied.end())});
+            },
+            [&] (const binary_operator& oper) {
+                return std::visit(overloaded_functor{
+                        [&] (const column_value& col) { return expr; },
+                        [&] (const std::vector<column_value>& cvs) { return expr; },
+                        [&] (const token&) { 
+                            return expression(binary_operator{column_value{new_cdef}, oper.op, oper.rhs});
+                        },
+                    }, oper.lhs);
+            },
+        }, expr);
+}
+
```

Comment:
I see that replace_xxx() is a popular function. I don't suggest to do it now, but if we collect a few more we can introduce a generic replace_term<Term> that will look for a class type down the tree and replace it.